crates/holochain_state/src/query/chain_head.rs

```diff
@@ -142,7 +141,10 @@ mod tests {
         })
         .collect();
 
+        // Other actions have a different author, so the 9th action should be the head for our author's chain
         let expected_head = shhs[8].clone();
+        // Shuffle so the head will sometimes be in scratch and sometimes be in the database and no always the last action by our author.
+        shhs.shuffle(&mut thread_rng());
```

Comment:
The way the test was written
- The chain head would always be in the scratch space. That's realistic but not testing the database query.
- The chain head would always be last once a filter on the author was done, which doesn't test the `fold` logic of the query properly by sometimes finding the data out of order. Again, it was realistic but not checking behaviour as broadly as it could